src/Resources/contao/classes/DataContainer.php

```diff
@@ -1027,4 +1071,90 @@ protected function getPickerInputField($value, $attributes='')
 	 * @throws \Exception
 	 */
 	abstract protected function save($varValue);
+
+
+	/**
+	 * Create a data container object in the picker environment
+	 *
+	 * @return DataContainer
+	 */
+	private function getDataContainer()
+	{
+		$this->validatePickerParams();
+
+		$do = \Input::get('do');
+		$table = \Input::get('table');
+		$id = \Input::get('id');
+		$act = \Input::get('act');
+		$rt = \Input::get('rt');
+
+		\Input::setGet('do', $this->strPickerDo);
+		\Input::setGet('table', $this->strPickerTable);
+		\Input::setGet('id', $this->intPickerId);
+		\Input::setGet('act', 'edit');
+		\Input::setGet('rt', REQUEST_TOKEN);
+
+		\Controller::loadDataContainer($this->strPickerTable);
+
+		$strDriver = 'DC_'.$GLOBALS['TL_DCA'][$this->strPickerTable]['config']['dataContainer'];
+
+		$objDca = new $strDriver($this->strPickerTable);
+		$objDca->id = $this->intPickerId;
+		$objDca->field = $this->strPickerField;
+
+		\Input::setGet('do', $do);
+		\Input::setGet('table', $table);
+		\Input::setGet('id', $id);
+		\Input::setGet('act', $act);
+		\Input::setGet('rt', $rt);
+
+		return $objDca;
+	}
+
+
+	/**
+	 * Validate the picker parameters
+	 *
+	 * @throws \RuntimeException
+	 */
+	private function validatePickerParams()
+	{
+		$arrModule = array();
+
+		foreach ($GLOBALS['BE_MOD'] as &$arrGroup)
+		{
+			if (isset($arrGroup[$this->strPickerDo]))
+			{
+				$arrModule =& $arrGroup[$this->strPickerDo];
+				break;
+			}
+		}
+
+		$this->import('BackendUser', 'User');
+
+		// Dynamically add the "personal data" module (see #4193)
+		if ($this->strPickerDo == 'login')
```

Comment:
This hardcoded stuff here smells to me. If the core needs this, somebody else might need it too?